### PR TITLE
I871 collapsible chart list

### PR DESCRIPTION
**Pull request by Serhiy91**

### CODE REVIEW

File: app/imports/ui/react/components/LoadableDatePicker.js [hunk 1]
```diff
@@ -28,6 +30,11 @@ const LoadableDatePicker = Loadable.Map({
         className="form-control"
         selected={selected ? moment(selected) : null}
         popperClassName={className}
+        onChange={(date) => {
+          if (onChange) {
+            onChange(date.startOf(startOf));
```
Comment (Dartv) [inline app/imports/ui/react/components/LoadableDatePicker.js]: Why?

File: app/imports/ui/react/components/timeline/helpers.js [hunk 2]
```diff
@@ -23,7 +24,31 @@ export const getScaleDates = (timeScale, partOfPastTime) => {
...
+  const lineInterval = lineEnd - lineStart;
+  const intervalToStartOfLine = lineStart - scaleStart;
+
+  return {
+    left: intervalToStartOfLine * 100 / scaleInterval,
+    width: lineInterval * 100 / scaleInterval,
+  };
+};
+
+export const getTimelineListProps2 = (scaleDates, lineStart) => {
```
Comment (Dartv) [inline app/imports/ui/react/components/timeline/helpers.js]: The names are super confusing. I should know what this function does by looking at the name

File: app/imports/ui/react/goals/components/ActionsMilestonesList.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      </span>
+    </ToggleButtonWrapper>
+
+    <Collapse {...{ isOpen }}>
+      {actionsMilestones.map(item => (
+        <TimelineListItem key={item.id} {...item} />
+      ))}
+
+      <ActionPanel>
+        <a href="">Add action</a>
```
Comment (Dartv) [inline app/imports/ui/react/goals/components/ActionsMilestonesList.js]: Why not use `Button` with `color="link"`?

File: app/imports/ui/react/goals/components/ActionsMilestonesList.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const ActionPanel = styled.div`
+  a {
+    margin-right: 15px;
+  }
+`;
+
+const enhance = compose(
+  withStateToggle(false, 'isOpen', 'toggle'),
+  withProps(({ goal }) => ({
+    actionsMilestones: reject(view(lenses.isCompleted), goal.points),
```
Comment (Dartv) [inline app/imports/ui/react/goals/components/ActionsMilestonesList.js]: You're just making this monolithic. What if we wanted to add risks to it? We should go back and name it `actionsMilestonesRisks`? This is bad. Why not just name it `items`, `entities` or something?

File: app/imports/ui/react/components/timeline/helpers.js (hunk 2 above)
Comment (Serhiy91) [reply app/imports/ui/react/components/timeline/helpers.js]: I have forgot to remove it

File: app/imports/client/apollo/utils/goals.js [hunk 5]
```diff
@@ -33,16 +33,16 @@ export const moveGoalWithinCacheAfterRestoring = curry((organizationId, goal, st
   }, store);
 });
 
-export const addActionToGoalFragment = curry((goalId, action) => {
+export const addActionToGoalFragment = curry((goalId, action, store = client) => {
```
Comment (Dartv) [inline app/imports/client/apollo/utils/goals.js]: I'm pretty sure default arguments mess up currying

Comment (Serhiy91) [reply app/imports/client/apollo/utils/goals.js]: @Dartv I also call  it from meteor env

**Comment by Serhiy91** [comment]
@Dartv fixed

**Pull request merged**